pkg/graph/resolvers/system_intake_system_test.go

```diff
@@ -83,3 +83,104 @@ func (s *ResolverSuite) TestIntakeRelatedSystems() {
 		s.Empty(data)
 	})
 }
+
+func (s *ResolverSuite) TestSystemIntakesByCedarSystemID() {
+	ctx := s.testConfigs.Context
+
+	var (
+		open1 uuid.UUID
+		open2 uuid.UUID
+
+		closed uuid.UUID
+	)
+
+	const (
+		system1 = "1"
+		system2 = "2"
+		system3 = "3"
+		system4 = "4"
+	)
+
+	s.Run("test getting system intakes by cedar system id", func() {
+		// create some intakes
+		intake1 := models.SystemIntake{
+			EUAUserID:   testhelpers.RandomEUAIDNull(),
+			RequestType: models.SystemIntakeRequestTypeNEW,
+			State:       models.SystemIntakeStateOpen,
+		}
+
+		create1, err := s.testConfigs.Store.CreateSystemIntake(ctx, &intake1)
+		s.NoError(err)
+		s.NotNil(create1)
+
+		open1 = create1.ID
+
+		intake2 := models.SystemIntake{
+			EUAUserID:   testhelpers.RandomEUAIDNull(),
+			RequestType: models.SystemIntakeRequestTypeNEW,
+			State:       models.SystemIntakeStateOpen,
+		}
+
+		create2, err := s.testConfigs.Store.CreateSystemIntake(ctx, &intake2)
+		s.NoError(err)
+		s.NotNil(create2)
+
+		open2 = create2.ID
+
+		intake3 := models.SystemIntake{
+			EUAUserID:   testhelpers.RandomEUAIDNull(),
+			RequestType: models.SystemIntakeRequestTypeNEW,
+			State:       models.SystemIntakeStateClosed,
+		}
+
+		create3, err := s.testConfigs.Store.CreateSystemIntake(ctx, &intake3)
+		s.NoError(err)
+		s.NotNil(create3)
+
+		closed = create3.ID
+
+		// link all systems to all system intakes
+		systemNumbers := []string{
+			system1,
+			system2,
+			system3,
+			system4,
+		}
+
+		err = sqlutils.WithTransaction(ctx, s.testConfigs.Store, func(tx *sqlx.Tx) error {
+			return s.testConfigs.Store.SetSystemIntakeSystems(ctx, tx, open1, systemNumbers)
+		})
+		s.NoError(err)
+
+		err = sqlutils.WithTransaction(ctx, s.testConfigs.Store, func(tx *sqlx.Tx) error {
+			return s.testConfigs.Store.SetSystemIntakeSystems(ctx, tx, open2, systemNumbers)
+		})
+		s.NoError(err)
+
+		err = sqlutils.WithTransaction(ctx, s.testConfigs.Store, func(tx *sqlx.Tx) error {
+			return s.testConfigs.Store.SetSystemIntakeSystems(ctx, tx, closed, systemNumbers)
+		})
+		s.NoError(err)
```

Comment:
nitpick, but these could all be in the same `WithTransaction` call, right?

```suggestion
		err = sqlutils.WithTransaction(ctx, s.testConfigs.Store, func(tx *sqlx.Tx) error {
		    if err = s.testConfigs.Store.SetSystemIntakeSystems(ctx, tx, open1, systemNumbers); err != nil {
		        return err
		    }
			if err = s.testConfigs.Store.SetSystemIntakeSystems(ctx, tx, open2, systemNumbers); err != nil {
		        return err
		    }
			if err = s.testConfigs.Store.SetSystemIntakeSystems(ctx, tx, closed, systemNumbers); err != nil {
		        return err
		    }
		    return nil
		})
		s.NoError(err)
```

Reply:
they could be sure, but i wanted to keep it separate for tests (also easier to know which err occurred, for example)